lua/ge/extensions/UI.lua

```diff
@@ -153,10 +153,24 @@ end
 --- This function is used to update the edit/spawn queue values for the UI indicator.
 -- @param spawnCount number
 -- @param editCount number
-local function updateQueue( spawnCount, editCount)
+-- @param queuedPlayers table
+local function updateQueue( spawnCount, editCount, queuedPlayers)
 	UIqueue = {spawnCount = spawnCount, editCount = editCount}
 	UIqueue.show = spawnCount+editCount > 0
 	sendQueue()
+
+    
+    if queuedPlayers ~= nil and queuedPlayers ~= {} then 
+        for key, player in pairs(MPVehicleGE.getPlayers()) do
+            if tableContains(queuedPlayers, key) then
+                guihooks.trigger("setQueueState", {playerID = key, state = true})
+            else
+                guihooks.trigger("setQueueState", {playerID = key, state = false})
+            end
+        end
+    else 
+        guihooks.trigger("resetQueueState")
+    end
```

Comment:
please refactor this code to use the existing `setQueue` trigger, it is fine to update the entire queue state when an event gets applied, as other data probably changes too
```suggestion
	UIqueue = {spawnCount = spawnCount, editCount = editCount, queuedPlayers = (queuedPlayers or {})}
	UIqueue.show = spawnCount+editCount > 0
	sendQueue()
```